client/internal/wgproxy/proxy_userspace.go

```diff
@@ -20,6 +20,7 @@ type WGUserSpaceProxy struct {
 
 // NewWGUserSpaceProxy instantiate a user space WireGuard proxy
 func NewWGUserSpaceProxy(wgPort int) *WGUserSpaceProxy {
+	log.Debugf("instantiate new userspace proxy")
```

Comment:
we should close the conn when calling Free for userspace proxy too.

Reply:
You are right but not exactly. Behind the eBPF proxy and the user space proxy has different life circle logic. In case of eBPF we instantiate it in a very early state but of the user space proxy does not have a global state. We instantiate a new proxy when we create a peer conn. And we free it up when we close the peer conn.
So basically for the Free() function make no sense in case of user space proxy and CloseConn() function make no sense in case of eBPF proxy. I left comments in the code at the relevant places.
```
// Free doing nothing because this implementation of proxy does not have global state
```

Reply:
Got it, thanks for clarifying.